RHOAI-16971 LAB-tuning - admin (#765)
opendatahub-io/opendatahub-documentation: assemblies/enabling-lab-tuning.adoc

LAB-tuning depends on several {productname-short} components working together to support model customization. It uses data science pipelines to run the tuning workflow, KServe to deploy and serve the teacher and judge models, and the Training Operator to run distributed model training across GPU-enabled nodes. LAB-tuning also relies on the model registry to manage model versions, storage connections (such as S3 or OCI) to store pipeline artifacts and model outputs, and GPU hardware profiles to schedule training workloads.

To enable LAB-tuning, an {openshift-cluster} cluster administrator must configure the required infrastructure and platform components by completing the following tasks:

* Install the required Operators
* Install the required components
* Configure a storage class that supports dynamic provisioning

A cluster administrator or a {productname-short} administrator must perform additional setup within the {productname-short} dashboard:

* Make LAB-tuning features visible in the dashboard
* Create a model registry
* Create a GPU hardware profile

== Requirements for LAB-tuning

* You have an {openshift-platform} cluster with cluster administrator access.

> * Your {openshift-platform} cluster includes at least one node with 4 GPUs (such as NVIDIA A100s) to run distributed LAB-tuning workloads.

Reviewer: ```suggestion
* Your {openshift-platform} cluster includes at least one node with 1 NVIDIA GPUs (e.g NVIDIA L40S 48 GB) for the LAB-tuning process. Multiple GPUs on the same or different nodes are required to run distributed LAB-tuning workloads.
```

> * Your {openshift-platform} cluster has additional GPU-enabled nodes to serve the teacher and judge models based on the selected model size and serving runtime.

Reviewer: ```suggestion
* Your {openshift-platform} cluster has additional GPU-enabled nodes to serve the [teacher and judge models](https://github.com/opendatahub-io/ilab-on-ocp/blob/main/manifests/README.md#deploying-teacher-and-judge-models) based on the selected model size and serving runtime.
```

Author: Do we know the specific requirements for the current "supported" teacher and judge models?

Reviewer: This could be documented in the linked doc

* Your environment meets the prerequisites for installing the required Operators and using the required components, storage, model registry, and GPU hardware profiles.

== Installing the required Operators for LAB-tuning

To enable your data scientists to customize models using LAB-tuning in {productname-short}, a cluster administrator must install the following Operators in {openshift-platform}:

* {productname-long} Operator

> * {org-name} Authorino Operator (Technical Preview)

Reviewer: According to the Release Notes draft for RHOAI 2.20 available at [RHOAIENG-21561](https://issues.redhat.com/browse/RHOAIENG-21561) now the recommended version is the stable one

```suggestion
* {org-name} Authorino Operator
```

.Prerequisites
* You are logged in to {productname-short} as a user with administrator privileges.
* The model registry component is enabled for your environment.

.Procedure
. Follow the steps described in link:https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2.19/html/managing_model_registries/creating-a-model-registry_managing-model-registries[Creating a model registry].

== Creating a GPU hardware profile for LAB-tuning

A GPU hardware profile is required to run LAB-tuning workloads in {productname-short}. LAB-tuning uses distributed training that must be scheduled on nodes with GPU resources. A GPU hardware profile allows users to target specific GPU-enabled worker nodes when launching pipelines, ensuring that training workloads run on compatible hardware.

You must configure a GPU hardware profile in {productname-short} that users can select when launching a LAB-tuning run.

.Prerequisites
* You are logged in to {productname-short} as a user with administrator privileges.
* The relevant hardware is installed and you have confirmed that it is detected in your environment.

.Procedure

> . Follow the steps described in link:https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2.19/html/working_with_accelerators/working-with-hardware-profiles_accelerators#creating-a-hardware-profile_accelerators[Creating a hardware profile] to create a GPU hardware profile with the following configurations:

Reviewer: ```suggestion
. Follow the steps described in link:https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2.19/html/working_with_accelerators/working-with-hardware-profiles_accelerators#creating-a-hardware-profile_accelerators[Creating a hardware profile] to create a _LAB-tuning_ hardware profile, adapting the following configurations to your specific cluster setup:
```

+
[cols="1,2", options="header"]
|===
| Setting
| Value

> 

Reviewer: ```suggestion

| CPU
| Default: 4 Cores; Minimum allowed: 2 Cores; Maximum allowed: 4 Cores

```

* Your {openshift-platform} cluster has at least one node with 1 NVIDIA GPUs (for example, NVIDIA L40S 48 GB) for the LAB-tuning process. Multiple GPUs on the same or different nodes are required to run distributed LAB-tuning workloads.
* To deploy the teacher model (link:https://catalog.redhat.com/software/containers/rhelai1/modelcar-mixtral-8x7b-instruct-v0-1/67922f1e167e94db874af7ab[rhelai1/modelcar-mixtral-8x7b-instruct-v0-1:1.4]), the {openshift-platform} cluster has a worker node with one or multiple GPUs capable of running the model (for example, a2-ultragpu-4g, which contains 4 x NVIDIA A100 with 80GB vRAM) and has at least 100 GiB of available disk storage to store the model.
* To deploy the judge model (link:https://catalog.redhat.com/software/containers/rhelai1/modelcar-prometheus-8x7b-v2-0/67922f21a4baf873b6f43d8c[rhelai1/modelcar-prometheus-8x7b-v2-0:1.4]), the {openshift-platform} cluster has a worker node with one or multiple GPUs capable of running this model (for example, a2-ultragpu-2g, which contains 2 x NVIDIA A100 with 80GB vRAM) and has at least at least 100 GiB of available disk storage to store the model.

> * Your environment meets the prerequisites for installing the required Operators and using the required components, storage, model registry, and GPU hardware profiles.

Reviewer: It looks good to me. Thanks!
